clients/pubsub-common/src/main/java/org/eclipse/hono/client/pubsub/PubSubBasedAdminClientManagerFactoryImpl.java

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.client.pubsub;
+
+import java.util.Objects;
+
+import com.google.api.gax.core.CredentialsProvider;
+
+import io.vertx.core.Vertx;
+import jakarta.enterprise.context.ApplicationScoped;
+
+/**
+ * A factory to create PubSubBasedAdminClientManagerImpl instances.
+ */
+@ApplicationScoped
+public class PubSubBasedAdminClientManagerFactoryImpl implements PubSubBasedAdminClientManagerFactory {
```

Comment:
what's the purpose of having this factory? It doesn't seem to do anything but instantiate the real PubSubBasedAdminClientManager, which could be done directly by the users of this factory, or am I mistaken?

Reply:
No, you are right. It doesn't do anything here. I just need it for the Device Communication API in the hono-extras repository, so I just moved it over there.